streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@arunmahadevan @satishd @harshach I spoke too soon. Having thought more about it, i feel that Optional.empty might introduce some regression. Regression would be from other processors like Aggregate, Join, Rule, etc. Some of the them use storm sql. Behaviour of storm sql or other processors is more likely to break anything if we use Optional.empty rather than what we currently have. Moreover, the current PR has also been tested in a SAM application that first brought this issue to light. I think for this release, I prefer the way it is in this PR - i.e. no entry for nullable fields in StreamlineEvent. Not disagreeing with any of the discussion here. And it could still cause some issues but better than Optional.empty. We need more thinking through and testing to make figure out the contract for nullable fields from avro schema flowing in SAM application